### PR TITLE
Tests for #20956

### PR DESCRIPTION
**Pull request by schrotthaufen**
Uses all --with-sysFOO flags

- [ ] Mark this if you consider it ready to merge
- [x] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

I've added building r2 with all the `--with-sys...` configure flags.

### CODE REVIEW

**Comment by trufae** [comment]
i guess that capstone3 support must go away

**Comment by schrotthaufen** [comment]
Ubuntu 22.04 ships capstone4, if that's an option for the build image. Alternatively, I could just remove `--with-syscapstone`, but that feels a bit like missing the point.

**Comment by trufae** [comment]
Capatone 3 and 4 should be fine for r2. But if you want accuracy or at least you work on latest x64 or arm64 binaries where you need pac instructions and other stuff added for the spectra then you can only use capstone5 which is not released.

i will take a look and see whats up with that option. Thanks for the pr! Stuff like this must be tested in every commit :)

**Comment by trufae** [comment]
Should be fixed after merging this https://github.com/radareorg/radare2/pull/20995 can you rebase when those commits hit the master branch?

**Comment by trufae** [comment]
Looks like its ssl time now

File: .github/workflows/tcc.yml [hunk 1]
```diff
@@ -131,3 +131,50 @@ jobs:
...
+      with:
+          repository: radareorg/radare2-testbins
+          path: test/bins
+    - name: Configure, build and install
+      env:
+          CC: tcc
+      run: |
+          sudo apt-get --assume-yes install libxxhash-dev liblz4-dev libzip-dev libmagic-dev libcapstone-dev
+          ./configure --prefix=/usr --with-compiler=tcc \
+              --with-syscapstone \
```
Comment (trufae) [inline .github/workflows/tcc.yml:164]: maybe also add --with-oenssl ?

**Comment by trufae** [comment]
Should be fixed in https://github.com/radareorg/radare2/pull/21014, pls rebase the pr and retry after merged :) thank you

File: .github/workflows/tcc.yml [hunk 2]
```diff
@@ -131,3 +131,50 @@ jobs:
...
+
+    - name: Run tests
+      env:
+        PKG_CONFIG_PATH: /usr/lib/x86_64-linux-gnu/pkgconfig
+      run: |
+        r2 -v
+        r2r -v
+        export R2R_SKIP_ASM=1
+        export R2R_SKIP_ARCHOS=1
+        make -j -C test
```
Comment (trufae) [inline .github/workflows/tcc.yml:180]:
```suggestion
        make -j -C test/unit
```

Comment (trufae) [reply .github/workflows/tcc.yml:180]: If you build with any version of capstone that is not v5 with the custom patches from r2 it wont pass the testsuite. so i would just run the unit tests and merge :) all bugs fixed here. thanks for the pr!

**Pull request merged**